connector-packager/connector_packager/xsd_validator.py

```diff
@@ -358,3 +357,6 @@ def warn_file_specific_rules_tdr(path_to_file: Path):
     if not authentication_attr_exists:
         logger.warning("Warning: 'authentication' attribute is missing from "
                        "<connection-normalizer>/<required-attributes>/<attribute-list> in " + str(path_to_file) + ".")
+                       
+    if 'dbname' not in attribute_list:
```

Comment:
I think we want the same conditions as before, just as a warning instead of an error.

Additionally a bit about why: "Warning: connection-metadata 'database' enabled but 'dbname' is not in required-attributes list. Consider adding it if the value is used in connection-builder or connection-properties scripts"

Reply:
Updated the warning and moved the logic to `validate_file_specific_rules_tdr` since properties is not available in `warn_file_specific_rules_tdr` unless we want to change its parameters